File: backend/src/routers/root.py
from datetime import datetime, timezone

from fastapi import APIRouter

from src.database import check_connection

router = APIRouter()


@router.get("/")
async def root():
    return {
        "service": "fastapi-backend",
        "status": "running",
        "message": "Hello Brewnet (https://www.brewnet.dev)",
    }


@router.get("/health")
async def health():
    db_connected = await check_connection()
    return {
        "status": "ok",
        "timestamp": datetime.now(timezone.utc).isoformat(),
        "db_connected": db_connected,
    }

File: backend/src/routes/root.ts
import { Router, Request, Response } from "express";
import { checkConnection } from "../database";

const router = Router();

router.get("/", (_req: Request, res: Response) => {
  res.json({
    service: "express-backend",
    status: "running",
    message: "🍺 Brewnet says hello!",
  });
});

router.get("/health", async (_req: Request, res: Response) => {
  const dbConnected = await checkConnection();
  res.json({
    status: "ok",
    timestamp: new Date().toISOString(),
    db_connected: dbConnected,
  });
});

export default router;
